src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py

```diff
@@ -331,19 +331,26 @@ def process_2d_data_object(self, data_set, key, unit):
                 self.current_dataset.qx_data = data_set
             elif self.q_names.index(key) == 1:
                 self.current_dataset.qy_data = data_set
-        elif key in self.q_uncertainty_names or key in self.q_resolution_names:
-            if ((self.q_uncertainty_names[0] == self.q_uncertainty_names[1]) or
-                    (self.q_resolution_names[0] == self.q_resolution_names[1])):
-                # All q data in a single array
+        elif key in self.q_resolution_names:
+            if (len(self.q_resolution_names) == 1
+                    or (self.q_resolution_names[0]
```

Comment:
Shouldn't this be an `and` rather than `or`?
(also below)

Reply:
I think you might have missed the last line of the statement, `== self.q_resolution_names[1])):`.

Reply:
errr I just might....